### PR TITLE
DBZ-804 Supporting negative scale values in Oracle grammar

### PR DESCRIPTION
**Pull request by gunnarmorling**
https://issues.jboss.org/browse/DBZ-804

I've also filed https://github.com/antlr/grammars-v4/issues/1190 upstream.

### CODE REVIEW

File: debezium-core/src/main/java/io/debezium/data/VariableScaleDecimal.java [hunk 1]
```diff
@@ -78,13 +78,30 @@ public static Struct fromLogical(Schema schema, SpecialValueDecimal value) {
...
+    /**
+     * Converts a value from its logical format to its encoded format - a struct containing
+     * the scale of the number and a binary representation of the number.
+     *
+     * @param schema of the encoded value
+     * @param value the value or the decimal
+     *
+     * @return the encoded value
+     */
+    public static Struct fromLogical(Schema schema, BigDecimal decimalValue) {
```
[the remark this answers is not in the document]
Comment (gunnarmorling) [reply debezium-core/src/main/java/io/debezium/data/VariableScaleDecimal.java]: Reworked this commit to address this remark.

File: debezium-core/src/main/java/io/debezium/data/VariableScaleDecimal.java [hunk 2]
```diff
@@ -70,11 +71,25 @@ public static Schema optionalSchema() {
...
+        Objects.requireNonNull(decimalValue, "decimalValue may not be null");
+
         Struct result = new Struct(schema);
-        final BigDecimal decimalValue = value.getDecimalValue().orElse(null);
-        assert decimalValue != null : "Unable to encode special value";
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/data/VariableScaleDecimal.java]: Note I've replaced the assert call by `requireNonNull`. I generally find asserts not overly useful. If we really want to  be sure that the value isn't null, let's enforce it without relying on any runtime parameters like _-ea_.

**Comment by jpechane** [comment]
@gunnarmorling LGTM now, if CI passes the PR will be merged

**Comment by jpechane** [comment]
@gunnarmorling Applied, thanks!

**Pull request merged**